shotover-proxy/tests/opensearch_int_tests/mod.rs

```diff
@@ -0,0 +1,102 @@
+use opensearch::{
+    auth::Credentials,
+    cert::CertificateValidation,
+    http::Url,
+    http::{
+        response::Response,
+        transport::{SingleNodeConnectionPool, TransportBuilder},
+        StatusCode,
+    },
+    indices::IndicesExistsParts,
+    params::Refresh,
+    BulkOperation, BulkParts, OpenSearch, SearchParts,
+};
+use serde_json::{json, Value};
+use test_helpers::docker_compose::docker_compose;
+use test_helpers::run_command;
+
+pub async fn index_documents(client: &OpenSearch) -> Response {
+    let index = "posts";
+    let exists_response = client
+        .indices()
+        .exists(IndicesExistsParts::Index(&[index]))
+        .send()
+        .await
+        .unwrap();
+
+    if exists_response.status_code() == StatusCode::NOT_FOUND {
+        let mut body: Vec<BulkOperation<_>> = vec![];
+        for i in 1..=10 {
+            let op = BulkOperation::index(json!({"title":"OpenSearch"}))
+                .id(i.to_string())
+                .into();
+            body.push(op);
+        }
+
+        client
+            .bulk(BulkParts::Index(index))
+            .body(body)
+            .refresh(Refresh::WaitFor)
+            .send()
+            .await
+            .unwrap()
+    } else {
+        exists_response
+    }
+}
+
+#[tokio::test(flavor = "multi_thread")]
+async fn passthrough_standard() {
+    // This setting changes the maximum number of memory map areas a process may have.
+    // OpenSearch requires vm.max_map_count to be set higher to run.
+    // This command sets it temporarily higher (for the length of the user session), it is reset after rebooting.
+    if &run_command("sysctl", &["vm.max_map_count"]).unwrap() != "vm.max_map_count = 262144\n" {
+        run_command("sudo", &["sysctl", "-w", "vm.max_map_count=262144"]).unwrap();
```

Comment:
Is this is in response to an actual failure you encountered on your machine?
Or did you just do it because the docs recommended it?

Reply:
Docs recommended it and a failure on my machine. Did it occur for you?

Reply:
The pr as is hung on my machine, I'll investigate more tomorrow

Reply:
We can avoid needing root by running opensearch in a non-clustered way that disables production checks.
This should be all the config needed to enable this:
![image](https://github.com/shotover/shotover-proxy/assets/5120858/342a7b1b-09f6-4ae2-a25e-c6223945413e)
(And also remove the second instance)

On my machine without the above changes the container fails to start without vm.max_map_count set.
However when using the above changes it just works.

Having our tests halt and wait for a sudo password is really problematic.
In the future we may want to test clustered opensearch but we can deal with that problem when we need to.
A possible fix I'm thinking of could be to have our CI setup script set vm.max_map_count and then have the test just fail locally (with a nice instructive error message) if they dont have the correct vm.max_map_count set.